### DISCUSSION TITLE
Shared Audio Library (Important Suggestion)

### DISCUSSION THREAD

**Discussion by cast-play** [AzuraCast/AzuraCast#7029] [General]
Hello! Congratulations to everyone involved and who made this excellent Azuracast project. I'm really enjoying it!

An important suggestion is to have an option for the administrator to share their audio library, and for stations to be able to program (schedule) them in their auto dj playlists. But only the administrator has permission to delete, change or add new audio (media) in this option to share the ready-made audio library with stations (users). Stations (users) will be allowed to schedule in your playlist. This is great, as the administrator can facilitate broadcasters' programming by sharing a library of ready-to-air media.
The other functions are perfect.
Thanks!

**Comment by Moonbase59** [comment]
It’s all there, using basic Linux and Docker mechanisms. You can mount, on your AzuraCast host, your central repository, then mount that (or parts of it) into your miscellaneous station’s media, even read-only. The stations can then _still_ adjust things within their song database, even when the mount is read-only. AzuraCast respects parts of its media folders being read-only and remembers changes in the station database.

Best of all worlds!

Hint: Use `/var/azuracast/docker-compose.override.yml` for that, example:

```yaml
services:
  web:
    volumes:
      - /mnt/Musik/Tagged:/var/azuracast/stations/niteradio/media/Tagged:ro
      - /mnt/Musik/Other:/var/azuracast/stations/niteradio/media/Other
```

Here, the (sub-)folder `Tagged` of my central music storage (NFS-mounted, from a remote TrueNAS) is mounted _read-only_ into station `niteradio`’s media, and `Other` as read-write.

Mounted files and folders must have appropriate Linux permissions, of course.

Doing it this way (mounting _into_ a station’s media folder) has the advantage that each station can have their own media in addition, maybe recordings, user uploads, etc.

I’ve been operating stations in this "software-agnostic" manner for ever so many years, starting with Rivendell back-in-the-days, Liquidsoap, Airtime, mAirList, AzuraCast, you name it.

**Reply by cast-play** [reply]
I tested it and managed to configure it. Thank you very much for helping me! I really like the Azuracast panel!

I had a problem that takes up disk space. Is there a way to avoid taking up space on the station that is receiving the share on Azuracast? Because the files are from a remote folder. And is there an option that when the share is added with a new user, I don't need to restart the system with the commands:
cd /var/azuracast
docker compose down
docker compose up -d
I'm a beginner, I'm learning.
Thank you very much for helping!

**Comment by cast-play** [comment]
Hello! Thanks for the answer.

I'm a beginner in Linux, so I don't know how to do this right.
The suggestion is to have this in the Azuracast admin panel.

Thanks!
